[Doc] V3.4 Query feedback doc (#53785)
StarRocks/starrocks: docs/en/using_starrocks/query_feedback.md

```

**Cause** The issue may result from data skew. When the right table has many partitions and one of them contains disproportionately large amounts of data, the system may incorrectly estimate the row count of right table after predicates are applied.

**Solution** During query execution, the system compares the `input/output rows` and `statistics estimated rows` of Left Child and Right Child, generating tuning guides. After optimization, the Join method is adjusted from Broadcast Join to Shuffle Join.

### Case 3: Inefficient First-Phase Pre-aggregation Mode

**Symptom** For data with good aggregation potential, the `auto` mode of First-phase aggregation may only aggregate a small amount of local data, missing the opportunity for performance gains.

**Solution** During query execution, the system collects `Input/Output Rows` for both local and global aggregations. Based on historical data, it evaluates the potential of the aggregation columns. If the potential is significant, the system enforces the use of `pre_aggregation` mode in local aggregations, maximizing data aggregation in the first phase and improving overall query performance.

## Limitations

> - A tuning guide can only be used to the exact same query based on which it was generated. It is not applicable to queries with the same pattern but different parameters.

Reviewer: ```suggestion
- A tuning guide can only be used for the exact same query for which it was generated. It is not applicable to queries with the same pattern but different parameters.
```
